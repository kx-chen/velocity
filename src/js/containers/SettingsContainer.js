import "../../css/Settings.css";
import SettingsIcon from "../components/SettingsIcon";
import SettingsModal from "../components/SettingsModal";
import { getTimeFormat, setTimeFormat } from "../actions/settings";
import { setWallpaper } from "../utils/utils";

import { useEffect, useState } from "preact/hooks";
import { h } from "preact";
import { connect } from "react-redux";

export function SettingsContainer({ changeTimeFormat, timeFormat, dispatch }) {
  const [menuOpen, setMenuOpen] = useState(false);

  useEffect(() => {
    // TODO: refactor, hacky.
    window.onclick = function (event) {
      let modal = document.getElementById("myModal");
      if (event.target === document.getElementById("body-wrapper")) {
        modal.style.display = "none";
        setMenuOpen(false);
      }
    };
  }, []);

  // TODO: find if use of setMenuOpen down there is ok
  return (
    <div>
      <div className="left-bottom">
        <SettingsIcon onClick={() => setMenuOpen(!menuOpen)} />
      </div>
      <SettingsModal
        handleWallpaperChange={setWallpaper}
        closeMenu={() => setMenuOpen(false)}
        visible={menuOpen}
        changeTimeFormat={changeTimeFormat}
        timeFormat={timeFormat}
      />
    </div>
  );
}

function mapDispatchToProps(dispatch, ownProps) {
  return {
    changeTimeFormat: (timeFormat) => dispatch(setTimeFormat(timeFormat)),
    dispatch,
  };
}

function mapStateToProps(state, ownProps) {
  return {
    ...state,
    timeFormat: state.time_format,
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(SettingsContainer);
