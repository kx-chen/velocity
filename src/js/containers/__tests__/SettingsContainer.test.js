import { SettingsContainer } from "../SettingsContainer";
import { h } from "preact";
import { mount } from "enzyme";

describe("SettingsContainer", () => {
  test("Settings opens on click", () => {
    const wrapper = mount(<SettingsContainer />);

    wrapper.find("#settings-cog").simulate("click");

    expect(wrapper).toMatchSnapshot();
  });

  test("Settings closes on clicking x", () => {
    const wrapper = mount(<SettingsContainer />);

    wrapper.find("#settings-cog").simulate("click");
    wrapper.find(".close-icon").simulate("click");

    expect(wrapper).toMatchSnapshot();
  });

  test("Settings closes when clicking settings cog again", () => {
    const wrapper = mount(<SettingsContainer />);

    wrapper.find("#settings-cog").simulate("click");
    wrapper.find("#settings-cog").simulate("click");

    expect(wrapper).toMatchSnapshot();
  });
});
