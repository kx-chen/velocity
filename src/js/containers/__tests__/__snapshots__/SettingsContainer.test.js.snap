// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsContainer Settings closes on clicking x 1`] = `
<SettingsContainer>
  <div>
    <div
      className="left-bottom"
    >
      <SettingsIcon
        onClick={[Function]}
      >
        <a
          id="settings-cog"
          onClick={[Function]}
        >
          <img
            alt="Settings"
            height="35"
            src="../../icons/settings-24px.svg"
            width="35"
          />
        </a>
      </SettingsIcon>
    </div>
    <SettingsModal
      closeMenu={[Function]}
      handleWallpaperChange={[Function]}
      visible={false}
    >
      <div
        className="modal-content modal"
        id="myModal"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <a
          className="close-icon close"
          onClick={[Function]}
        >
          <img
            alt="Close"
            height="35"
            src="../../icons/close-black-18dp.svg"
            width="35"
          />
        </a>
        <h1
          className="settings"
        >
           Settings 
        </h1>
        <hr
          align="left"
          width="25%"
        />
        <h2>
          Set a background
        </h2>
        <input
          accept="image/*"
          id="file"
          onChange={[Function]}
          type="file"
        />
        <h2>
          Time Format
        </h2>
        <p>
          Default format is 24 hours.
        </p>
        <input
          id="format"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="format"
        >
          12 hour
        </label>
      </div>
    </SettingsModal>
  </div>
</SettingsContainer>
`;

exports[`SettingsContainer Settings closes when clicking settings cog again 1`] = `
<SettingsContainer>
  <div>
    <div
      className="left-bottom"
    >
      <SettingsIcon
        onClick={[Function]}
      >
        <a
          id="settings-cog"
          onClick={[Function]}
        >
          <img
            alt="Settings"
            height="35"
            src="../../icons/settings-24px.svg"
            width="35"
          />
        </a>
      </SettingsIcon>
    </div>
    <SettingsModal
      closeMenu={[Function]}
      handleWallpaperChange={[Function]}
      visible={false}
    >
      <div
        className="modal-content modal"
        id="myModal"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <a
          className="close-icon close"
          onClick={[Function]}
        >
          <img
            alt="Close"
            height="35"
            src="../../icons/close-black-18dp.svg"
            width="35"
          />
        </a>
        <h1
          className="settings"
        >
           Settings 
        </h1>
        <hr
          align="left"
          width="25%"
        />
        <h2>
          Set a background
        </h2>
        <input
          accept="image/*"
          id="file"
          onChange={[Function]}
          type="file"
        />
        <h2>
          Time Format
        </h2>
        <p>
          Default format is 24 hours.
        </p>
        <input
          id="format"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="format"
        >
          12 hour
        </label>
      </div>
    </SettingsModal>
  </div>
</SettingsContainer>
`;

exports[`SettingsContainer Settings opens on click 1`] = `
<SettingsContainer>
  <div>
    <div
      className="left-bottom"
    >
      <SettingsIcon
        onClick={[Function]}
      >
        <a
          id="settings-cog"
          onClick={[Function]}
        >
          <img
            alt="Settings"
            height="35"
            src="../../icons/settings-24px.svg"
            width="35"
          />
        </a>
      </SettingsIcon>
    </div>
    <SettingsModal
      closeMenu={[Function]}
      handleWallpaperChange={[Function]}
      visible={true}
    >
      <div
        className="modal-content modal"
        id="myModal"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <a
          className="close-icon close"
          onClick={[Function]}
        >
          <img
            alt="Close"
            height="35"
            src="../../icons/close-black-18dp.svg"
            width="35"
          />
        </a>
        <h1
          className="settings"
        >
           Settings 
        </h1>
        <hr
          align="left"
          width="25%"
        />
        <h2>
          Set a background
        </h2>
        <input
          accept="image/*"
          id="file"
          onChange={[Function]}
          type="file"
        />
        <h2>
          Time Format
        </h2>
        <p>
          Default format is 24 hours.
        </p>
        <input
          id="format"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="format"
        >
          12 hour
        </label>
      </div>
    </SettingsModal>
  </div>
</SettingsContainer>
`;
